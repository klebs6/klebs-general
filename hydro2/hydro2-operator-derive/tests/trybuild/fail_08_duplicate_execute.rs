// ---------------- [ File: hydro2-operator-derive/tests/trybuild/fail_08_duplicate_execute.rs ]
//! Specifying `execute` twice â†’ compile_fail.
use hydro2_operator_derive::*;
use hydro2_operator::*; 
use named_item_derive::*;
use named_item::*;

#[derive(Debug,NamedItem,Operator)]
#[operator(
    execute="fn1",
    execute="fn2", // should fail
    opcode="BasicOpCode::TestOp"
)]
pub struct DuplicateExecute {
    name: String,
}

impl DuplicateExecute {
    pub async fn fn1(&self) -> NetResult<()> {
        Ok(())
    }
    pub async fn fn2(&self) -> NetResult<()> {
        Ok(())
    }
}

fn main() {}

