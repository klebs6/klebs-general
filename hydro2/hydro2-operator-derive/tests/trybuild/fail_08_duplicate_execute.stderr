error: Execute key was provided more than once in `#[operator(...)]`.
  --> tests/trybuild/fail_08_duplicate_execute.rs:11:5
   |
11 |     execute="fn2", // should fail
   |     ^^^^^^^
